File: Tellma/Controllers/DTO/EntitiesResponse.cs
﻿using Tellma.Entities;
using System.Collections.Generic;

namespace Tellma.Controllers.Dto
{
    /// <summary>
    /// Represents a response of raw entities for save
    /// </summary>
    public class EntitiesResponse<TEntity> where TEntity : Entity
    {
        public Dictionary<string, object> Extras { get; set; }

        public IEnumerable<TEntity> Result { get; set; }

        public string CollectionName { get; set; }

        public Dictionary<string, IEnumerable<Entity>> RelatedEntities { get; set; }

        public bool IsPartial { get; set; }
    }
}

File: Tellma.Api.Web/ClientApp/src/app/data/dto/entities-response.ts
import { EntityWithKey } from '../entities/base/entity-with-key';
import { Entity } from '../entities/base/entity';
import { Collection } from '../entities/base/metadata';

export interface EntitiesResponse<TEntity extends Entity = EntityWithKey> {
  Extras: {
    [key: string]: any;
  };
  Result: TEntity[];
  CollectionName: Collection;
  RelatedEntities: {
    [key: string]: EntityWithKey[];
  };
  ServerTime: string;
}
